gammapy/astro/darkmatter/utils.py

```diff
@@ -52,3 +66,181 @@ def compute_jfactor(self):
+
+    Parameters
+    ----------
+    dataset : `~gammapy.cube.fit.MapDataset`
+        Simulated dark matter annihilation dataset.
+    masses : list of `~astropy.units.Quantity`
+        List of particle masses where the values of :math:`\sigma\nu` will be calculated.
+    channels : list of strings allowed in `~gammapy.astro.darkmatter.PrimaryFlux`
+        List of channels where the values of :math:`\sigma\nu` will be calculated.
+    jfact : `~astropy.units.Quantity` (optional)
+        Integrated J-Factor needed when `~gammapy.image.models.SkyPointSource` spatial model is used, default value 1.
+    absorption_model : `~gammapy.spectrum.models.Absorption` (optional)
+        Absorption model, default is None.
+    z: float (optional)
+        Redshift value, default value 0.
+    k: int (optional)
+        Type of dark matter particle (k:2 Majorana, k:4 Dirac), default value 2.
+    xsection: `~astropy.units.Quantity` (optional)
+        Thermally averaged annihilation cross-section, default value declared in `~gammapy.astro.darkmatter.DMAnnihilation`.
+
+    Examples
+    --------
+    This is how you may run the `SigmaVEstimator`::
+
+        import logging
+        logging.basicConfig()
+        logging.getLogger("gammapy.astro.darkmatter.utils").setLevel("INFO")
+
+        jfact = 3.41e19 * u.Unit("GeV2 cm-5")
+        channels = ["b", "t", "Z"]
+        masses = [70, 200, 500, 5000, 10000, 50000, 100000]*u.GeV
+        estimator = SigmaVEstimator(simulated_dataset, masses, channels, jfact=JFAC)
+        result = estimator.run()
+    """
+
+    def __init__(
+        self,
+        dataset,
+        masses,
+        channels,
+        jfact=1,
+        absorption_model=None,
+        z=0,
+        k=2,
+        xsection=None,
+    ):
+
+        self.dataset = dataset
+        self.masses = masses
+        self.channels = channels
+        self.jfact = jfact
+        self.absorption_model = absorption_model
+        self.z = z
+        self.k = k
+
+        if not xsection:
+            xsection = DMAnnihilation.THERMAL_RELIC_CROSS_SECTION
+        self.xsection = xsection
+
+        self._spatial_model = dataset.model.spatial_model
+        self._geom = dataset.counts.geom
+        self._exposure = dataset.exposure
+        self._background_model = dataset.background_model
+        self._psf = dataset.psf
+        self._edisp = dataset.edisp
+
+        self._counts_map = WcsNDMap(self._geom, np.random.poisson(dataset.npred().data))
+
+    def run(self, optimize_opts=None, covariance_opts=None):
+        """Run the SigmaVEstimator for all channels and masses.
+
+        Parameters
+        ----------
+        optimize_opts : dict
+            Options passed to `Fit.optimize`.
+        covariance_opts : dict
+            Options passed to `Fit.covariance`.
+
+        Returns
+        -------
+        result : dict
+            Dict with results as `~astropy.table.Table` objects for each channel.
+        """
+
+        result = {}
```

Comment:
Would it make sense to use a flattened data structure here? I.e. use an `astropy.table.Table` with a `mass` and `channel` column? For every combination of `channel` and `mass` one row is added to the table. I think this simplifies the code as well as the data structure that is returned.

Reply:
There is a certain hierarchy to be kept in the results (at least from the user-access point of view) where every channel has a list of (mass, sigma-v, err, etc.) values. This is the main reason why tables are linked to dictionary items (channels).